c/datasetjson.c

```diff
@@ -1671,7 +1669,117 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
   finishResponse(response);     
 #endif /* __ZOWE_OS_ZOS */
 }
+//new function
+
+static int getDSCB(char* datasetName, char* dscb, int* dscbLength){
```

Comment:
The point of having the length was to do a check to make sure the caller didn't call it with a bad array size.

`static int getDSCB(char *datasetName, char *dscb, int dscbLength) {
   if (dscbLength < INDEX_DCSB) {
     zowelog("");
     respondWithJsonError(internal server error);
     return 0;
   }
}`

Something like this.